### PR TITLE
Add option to filter builds/test from comment by a given statuses

### PR DESCRIPTION
**Pull request by nikromen**
Just to add to code more reusability - if we may want to filter models by their different statuses in some CommentEvent - you can now filter them by a given statuses.

And also the main point of this PR - I added some logs so I can see which targets passed to `targets_override` in logs because I am clueless when debugging #1375 :D 

---

RELEASE NOTES BEGIN
￼
RELEASE NOTES END

### CODE REVIEW

**Review by lbarcziova — COMMENTED** [review]
The `FAILED_MODELS` constant looks a little bit fragile to me, could we use already existing and used in the implementation [TestingFarmResult](https://github.com/packit/packit-service/blob/9b26d91a8eeaa877a8ca351c67c02898d2b608c8/packit_service/models.py#L1494) enum states for TF and [PG_BUILD_STATUS_FAILURE](https://github.com/packit/packit-service/blob/9b26d91a8eeaa877a8ca351c67c02898d2b608c8/packit_service/constants.py#L55) for the build?

Also now looking into PG_BUILD_STATUS_FAILURE=`failure` and `failed` used in the implementation for rebuilding failed, couldn't this also cause the bugs with `packit rebuild-failed` ? :D (but since the retesting failed also doesn't work, there is probably still also something else wrong)

**Comment by nikromen** [comment]
> Also now looking into PG_BUILD_STATUS_FAILURE=failure and failed used in the implementation for rebuilding failed, couldn't this also cause the bugs with packit rebuild-failed ?

Hmm yes that's another bug :D thank you! but it's not the current problem since the result should be opposite - it should rebuild less builds... not more (everything) :D

**Comment by lbarcziova** [comment]
> Hmm yes that's another bug :D thank you! but it's not the current problem since the result should be opposite - it should rebuild less builds... not more (everything) :D

But as we already discussed, if you create a new copr build task where `build_targets_override` is None (because there are no builds with `failed` status, only with `failure`), builds will be run normally, for each target in the config.

**Comment by nikromen** [comment]
Yes but also as we discussed user probably won't use rebuild-failed to all passed builds. Now we don't have functionality in `targets_override` to basically "skip" the building part. So the solution can be that overriding can be done with empty set:
```py
if targets_override is not None:
    override here
```
so we can override it with empty set. But as far as I tested - this still isn't an issue. I think that `targets_override` aren't even called (it's behaving like normal /packit build) - that's why I am adding these logs to code :D

**Comment by lbarcziova** [comment]
> Now we don't have functionality in targets_override to basically "skip" the building part.

Yes, because it doesn't really make sense to create copr build handler task if we don't want to run any builds...

> Yes but also as we discussed user probably won't use rebuild-failed to all passed builds. 

That's still true. But since there is a bug and no builds are found in the DB because of wrong status, this scenario happens.

>  I think that targets_override aren't even called (it's behaving like normal /packit build) - that's why I am adding these logs to code :D

That's what I wrote :D it behaves like normal /packit build because there are no builds with 'failed' status found in the DB, therefore the build_targets_override is empty.

**Comment by nikromen** [comment]
> That's still true. But since there is a bug and no builds are found in the DB because of wrong status, this scenario happens.

oh, now I get it :D makes sense, so can you please review the code once more so we can try it? :innocent:

File: packit_service/constants.py
Comment (lbarcziova) [inline packit_service/constants.py:215]: I meant these to use directly in the code but will leave it up to you (but I can see from the tests it probably causes circular imports).

**Review by lbarcziova — APPROVED** [review]
LGTM once tests are passing

**Pull request merged**
